<2주차 과제 1 - Counter 앱 만들고 파일 분리하기>

xunxee opened the pull request:
1. 미완성의 첫 번째 commit 입니다.
2. 기본적인 셋팅을 진행했습니다.
  - App, Title, ClickButton, NumberButtons 컴포넌트를 만들어 App.jsx에 import 해주었습니다.
3. 파일 설명
  - App.jsx:
      useState를 만들어 컴포넌트들에 count 데이터를 전달합니다.
  - Title.jsx:
      "Counter" 라는 제목을 위한 컴포넌트입니다.
  - ClickMeButton.jsx:
      "Click Me" 버튼을 위한 컴포넌트입니다.
      현재 App.jsx에서 생성된 'count' 데이터를 porps로 전달한 상태입니다.
  - NumberButtons.jsx:
      숫자 버튼을 위한 컴포넌트입니다.

xunxee committed:
<2주차 과제 1 Counter 앱 만들고 파일 분리하기 -> 첫번째 제출>

1. 미완성의 첫 번째 commit 입니다.
2. 기본적인 셋팅을 진행했습니다.
  - App, Title, ClickButton, NumberButtons 컴포넌트를 만들어 App.jsx에 import 해주었습니다.
3. 파일 설명
  - App.jsx:
      useState를 만들어 컴포넌트들에 count 데이터를 전달합니다.
  - Title.jsx:
      "Counter" 라는 제목을 위한 컴포넌트입니다.
  - ClickMeButton.jsx:
      "Click Me" 버튼을 위한 컴포넌트입니다.
      현재 App.jsx에서 생성된 'count' 데이터를 porps로 전달한 상태입니다.
  - NumberButtons.jsx:
      숫자 버튼을 위한 컴포넌트입니다.

daadaadaah commented on src/App.jsx:
```diff
@@ -0,0 +1,24 @@
+import React, { useState } from 'react';
+import Title from './Title';
```
맥락 구분을 위해 한줄 띄어줍시당!ㅎㅎ

daadaadaah commented on src/ClickMeButton.jsx:
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
```
react 17 버전부터 이부분 생략 가능해요!
현재 리액트 버전 확인하시구~ 적용시켜보세요!
참고 : https://ko.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#whats-different-in-the-new-transform

xunxee replied:
확인했습니당

daadaadaah commented on src/index.jsx:
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
```
맥락 구분을 위해 한줄 띄어줍시다!ㅎㅎ

xunxee replied:
컴포넌트를 import 하기 전에, 맥락 구분 기억하겠습니다 ㅎㅎ🙏🏼

daadaadaah commented on src/ClickMeButton.jsx:
```diff
@@ -0,0 +1,16 @@
+  );
+}
+
+export default ClickMeButton;
```
컴포넌트 함수는 `export default function Button() { ... } ` 이렇게 한 줄로 표현 가능해요!

xunxee replied:
export default를 function 선언할 때 같이 사용할 수 있군요!🐥

daadaadaah commented on src/Title.jsx:
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+function Title() {
+  return (
+    <p>Counter</p>
+  );
+}
```
컴포넌트는 잘게 쪼개는 것보다 '왜' 나누는지가 더 중요해요!
Title 컴포넌트로 분리하는 것과 App에서 `<p>Counter</p>` 을 그대로 사용하는것
이 둘을 비교해서 어떤게 더 코드를 파악하기 쉬울까요? 한번 생각해보세요!

xunxee replied:
Counter 자체를 컴포넌트로 나누는 것이 더욱 코드를 복잡하게 만드는 것 같아요!
App에서 그래도 사용하는 편이 코드를 파악하는 것에 더 좋은 방법인 것 같습니당🙏🏼

daadaadaah reviewed (changes requested):
미완성이지만 PR 날려주신 점 너무 좋습니다!👍

다음 PR에서는 클릭 이벤트에 대한 기능 구현도 도전해보세요🚀

xunxee committed:
<2주차 과제1 - Counter 앱 만들고 파일 분리하기 -> 두번째 제출>

1. 리뷰를 반영했습니다.
  - 리액트 버전을 확인하고 import React from 'react'; 를 생략해주었습니다.
  - Counter 컴포넌트를 제거하고 App에 직접 넣어, 코드를 파악하기 쉽도록 바꾸었습니다.
2. 클릭 이벤트 기능을 구현했습니다.
  - App.jsx 안에 "handleClickMe" 함수와 "handleNumberButton" 함수를 작성하여
    porps로 전달하여 클릭 이벤트를 구현했습니다.

daadaadaah commented on src/App.jsx:
```diff
@@ -0,0 +1,36 @@
+      <p>Counter</p>
+      <ClickMeButton
+        count={count}
+        handleClickMe={handleClickMe}
```
저는 직접적으로 로직을 처리하는 이벤트 핸들러 함수의 경우 handleXXX 으로 함수 이름을 정하고,
중간에 전달하는 역할 정도만 하는 props 같은 이름으로 사용할 때는 onXXX 으로 사용하는 편이에요. 
왜냐하면, 실제 일을 `처리`하는 관점과 `사용`하는 입장에서 바라볼 때의 의도를 각각 드러내줄 수 있기 때문이예요! 같이 읽어보면 좋은 링크 하나 첨부해요~ 한번 읽어보세요!

참고:
- [https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364](https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364)

xunxee replied:
감사합니다! 이렇게 컨벤션을 따라서 해보니 훨씬 깔끔한 것 같아요!

daadaadaah commented on src/App.jsx:
```diff
@@ -0,0 +1,36 @@
+
+  const { count } = state;
+
+  function handleClickMe() {
+    setState({
+      count: count + 1,
+    });
+  }
+
+  function handleNumberButton(number) {
+    setState({
+      count: count + number,
+    });
+  }
```
1개의 함수를 사용해서 ClickMe 클릭 이벤트와 NumberButton 클릭 이벤트 둘다 관리할 수 있어요!ㅎㅎ
한번 도전해보세요~🚀

xunxee committed:
리뷰 반영("함수 네이밍 수정" 및 "클릭 이벤트 결합")

1. 함수 네이밍을 수정했습니다.
2. 기존에 2개의 관리되던 이벤트 핸들러를 1개로 결합했습니다.

daadaadaah reviewed (approved):
수고하셨습니다!👏
이번주 피드백 강의를 보며 아샬님은 어떻게 문제를 해결했는지 잘 관찰해보세요~!
